src/time_steppers.jl

```diff
@@ -45,237 +42,224 @@ time_step!(model; Nt, Δt, kwargs...) = time_step!(model, Nt, Δt; kwargs...)
-    boundary_condition_args = (model.clock.time, model.clock.iteration, U, Φ, model.parameters)
+    arch = model.architecture
+    grid = model.grid
+    bc_args = (model.clock.time, model.clock.iteration, U, C, model.parameters)
 
     # Pre-computations:
-    @launch device(arch) config=launch_config(grid, 3) store_previous_source_terms!(grid, Gⁿ, G⁻)
-    fill_halo_regions!(merge(U, Φ), bcs.solution, arch, grid, boundary_condition_args...)
+    store_previous_source_terms!(G⁻, arch, grid, Gⁿ)
+    fill_halo_regions!(merge(U, C), model.boundary_conditions.solution, arch, grid, bc_args...)
 
-    @launch device(arch) config=launch_config(grid, 3) calc_diffusivities!(K, grid, closure, buoyancy, U, Φ)
-    fill_halo_regions!(K, bcs.pressure, arch, grid) # diffusivities share bcs with pressure.
-    @launch device(arch) config=launch_config(grid, 2) update_hydrostatic_pressure!(p.pHY′, grid, buoyancy, Φ)
-    fill_halo_regions!(p.pHY′, bcs.pressure, arch, grid)
+    calculate_diffusivities!(K, arch, grid, model.closure, model.buoyancy, U, C)
+    fill_halo_regions!(K, model.boundary_conditions.pressure, arch, grid) # diffusivities share bcs with pressure.
+
+    @launch device(arch) config=launch_config(grid, 2) update_hydrostatic_pressure!(P.pHY′, grid, model.buoyancy, C)
+    fill_halo_regions!(P.pHY′, model.boundary_conditions.pressure, arch, grid)
 
     # Calculate tendency terms (minus non-hydrostatic pressure, which is updated in a pressure correction step):
-    calculate_interior_source_terms!(Gⁿ, arch, grid, coriolis, closure, U, Φ, p.pHY′, K, forcing, 
+    calculate_interior_source_terms!(Gⁿ, arch, grid, model.coriolis, model.closure, U, C, P.pHY′, K, model.forcing, 
                                      model.parameters, model.clock.time)
-    calculate_boundary_source_terms!(Gⁿ, arch, grid, bcs.solution, boundary_condition_args...)
+    calculate_boundary_source_terms!(Gⁿ, model.boundary_conditions.solution, arch, grid, bc_args...)
 
     # Complete explicit substep:
-    @launch device(arch) config=launch_config(grid, 3) adams_bashforth_update_source_terms!(grid, Gⁿ, G⁻, χ)
+    adams_bashforth_update_source_terms!(Gⁿ, arch, grid, χ, G⁻)
 
     # Start pressure correction substep with a pressure solve:
-    fill_halo_regions!(Gⁿ[1:3], bcs.tendency[1:3], arch, grid)
-    @launch device(arch) config=launch_config(grid, 3) calculate_poisson_right_hand_side!(arch, grid,
-                                                                                          model.poisson_solver.bcs,
-                                                                                          Δt, U, Gⁿ, RHS)
-    solve_for_pressure!(arch, model)
-    fill_halo_regions!(p.pNHS, bcs.pressure, arch, grid)
+    fill_halo_regions!(Gⁿ[1:3], model.boundary_conditions.tendency[1:3], arch, grid)
+    @launch device(arch) config=launch_config(grid, 3) calculate_poisson_right_hand_side!(model.poisson_solver.storage, arch, grid, 
```

Comment:
I wonder if defining macros like `@launch_3d = @launch device(arch) config=launch_config(grid, 3)`, `@launch_xy`, `@launch_xz`, etc. would make these kernels a bit more readable. Low priority as we're probably the only ones reading this code right now though.